src/components/common/Form.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react';
+import Autocomplete from 'react-autocomplete';
+
+class Form extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedAddress: '',
+      isCollectionScheduleTextHidden: true,
+      isAutoTextHidden: false,
+      dateFormat: 'D/M/YYYY',
+      scheduleType: {
+        trash: 'trash',
+        recycle: 'recycle',
+        leaf: 'leaf',
+      },
+      dayOfWeek: {
+        sunday: 0,
+        monday: 1,
+        tuesday: 2,
+        wednesday: 3,
+        thrusday: 4,
+        friday: 5,
+        saturday: 6,
+      },
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+  handleClick(event) {
+    this.setState({
+      isAutoTextHidden: false,
+      isCollectionScheduleTextHidden: true,
+      selectedAddress: '',
+    });
+  }
+  renderList(dataList) {
+    const selectedAddress = this.state.selectedAddress;
+    const items = dataList.map((item, index) => ({
```

Comment:
you probably don't need to loop through these items since you aren't doing anything special with them.

In your automcomplete you can just access item._id and item.address1 in place of id and label